### PR TITLE
fix: Lower case `BITCOIN:` URI

### LINKED ISSUE #16: Lower case `bitcoin:`
See https://github.com/rust-bitcoin/rust-bitcoin/issues/1843

This crate also upper-cases `BITCOIN:`, should lower it.

### PR DESCRIPTION
**Pull request by realeinherjar**
Closes #16.
I also changed the `Display` docstrings for `Uri`.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -56,7 +56,7 @@ use core::convert::{TryFrom, TryInto};
 use bitcoin::address::NetworkValidation;
 
 pub use de::{DeserializeParams, DeserializationState, DeserializationError};
-pub use ser::{SerializeParams};
+pub use ser::SerializeParams;
```
Comment (realeinherjar) [inline src/lib.rs:59]: `cargo clippy` fix

File: src/lib.rs [hunk 2]
```diff
@@ -73,8 +73,9 @@ pub use ser::{SerializeParams};
 /// `Display` is implemented for `Uri` so you can format it naturally. However it currently does
 /// **not** support alignment.
 ///
-/// To display the URI optimized for QR codes use alternate formatting (`{:#}`). The code assumes
-/// strict BIP-21, so it displays the schema as upper case. This is incompatible with some (buggy) wallets but creates the most optimal QR codes.
+/// The code does _not_ assume strict BIP-21, so it displays the schema lower case
+/// instead as upper case.
+/// This makes it compatible with some (buggy) wallets but does not create the most optimal QR codes.
```
Comment (realeinherjar) [inline src/lib.rs:78]: Feel free to edit this

File: src/ser.rs [hunk 3]
```diff
@@ -123,11 +123,7 @@ fn maybe_display_param(writer: &mut impl fmt::Write, key: impl fmt::Display, val
 #[rustfmt::skip]
 impl<'a, T> fmt::Display for Uri<'a, bitcoin::address::NetworkChecked, T> where for<'b> &'b T: SerializeParams {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        if f.alternate() {
```
Comment (realeinherjar) [inline src/ser.rs:126]: This was confusing. I could not find anywhere a `fn alternate` function.
I also looked into some deps but with no avail.
If there is a more elegant solution, please point me towards that.

Comment (DanGould) [reply src/ser.rs:126]: https://doc.rust-lang.org/std/fmt/struct.Formatter.html#method.alternate

@Kixunil  may we still want `bitcoin:{:#}` if `f.alternate()`?

File: src/lib.rs (hunk 1 above)
Comment (DanGould) [reply src/lib.rs:59]: I'm glad you pointed this out in a comment @realeinherjar. I like to see formatting changes in their own commit.

Comment (realeinherjar) [reply src/lib.rs:59]: Done!

File: src/ser.rs (hunk 3 above)
Comment (DanGould) [inline src/ser.rs:126]: I request these deleted lines from 126 are still included while lowercasing both conditional branches

**Review by DanGould — CHANGES_REQUESTED** [review]

File: src/ser.rs (hunk 3 above)
Comment (Kixunil) [reply src/ser.rs:126]: Yes, this is required by me too. It's only about schema, not other parts.

**Pull request closed without merging**